vnext/Microsoft.ReactNative/Modules/Animated/NativeAnimatedModule.cpp

```diff
@@ -16,6 +17,16 @@ NativeAnimatedModule::NativeAnimatedModule(Mso::CntPtr<Mso::React::IReactContext
   m_nodesManager = std::make_shared<NativeAnimatedNodeManager>(NativeAnimatedNodeManager());
 }
 
+NativeAnimatedModule::~NativeAnimatedModule() {
+  auto uiDispatcher =
+      winrt::Microsoft::ReactNative::implementation::ReactDispatcher::GetUIDispatcher(m_context->Properties());
+
+  // To make sure that we destroy UI components in UI thread.
+  if (!uiDispatcher.HasThreadAccess()) {
+    uiDispatcher.Post([maanger = std::move(m_nodesManager)]() {});
+  }
```

Comment:
>} [](start = 2, length = 1)

What if we don't have thread access?

Reply:
Then the destruction of m_nodesManager will happen on the UI thread already